tests/Upload/ResumableUploaderTest.php

```diff
@@ -0,0 +1,169 @@
+
+class ResumableUploaderTest extends \PHPUnit_Framework_TestCase
+{
+    private $requestWrapper;
+    private $stream;
+    private $successBody;
+
+    public function setUp()
+    {
+        $this->requestWrapper = $this->prophesize('Google\Cloud\RequestWrapper');
+        $this->stream = Psr7\stream_for('abcd');
+        $this->successBody = '{"canI":"kickIt"}';
+    }
+
+    public function testUploadsData()
+    {
+        $response = new Response(200, ['Location' => 'theResumeUri'], $this->successBody);
+
+        $this->requestWrapper->send(
+            Argument::type('Psr\Http\Message\RequestInterface'),
+            Argument::type('array')
+        )->willReturn($response);
+
+        $uploader = new ResumableUploader(
+            $this->requestWrapper->reveal(),
+            $this->stream,
+            'http://www.someuri.com'
+        );
+
+        $this->assertEquals(json_decode($this->successBody, true), $uploader->upload());
+    }
+
+    public function testGetResumeUri()
+    {
+        $resumeUri = 'theResumeUri';
+        $response = new Response(200, ['Location' => $resumeUri]);
+
+        $this->requestWrapper->send(
+            Argument::type('Psr\Http\Message\RequestInterface'),
+            Argument::type('array')
+        )->willReturn($response);
+
+        $uploader = new ResumableUploader(
+            $this->requestWrapper->reveal(),
+            $this->stream,
+            'http://www.someuri.com'
+        );
+
+        $this->assertEquals($resumeUri, $uploader->getResumeUri());
+    }
+
+    public function testResumesUpload()
+    {
+        $response = new Response(200, [], $this->successBody);
+        $statusResponse = new Response(200, ['Range' => 'bytes 0-2']);
+
+        $this->requestWrapper->send(
+            Argument::type('Psr\Http\Message\RequestInterface'),
+            Argument::type('array')
+        )->willReturn($response);
+
+        $this->requestWrapper->send(
+            Argument::that(function ($request) {
+                return $request->getHeaderLine('Content-Range') === 'bytes */*';
+            }),
+            Argument::type('array')
+        )->willReturn($statusResponse);
+
+        $uploader = new ResumableUploader(
+            $this->requestWrapper->reveal(),
+            $this->stream,
+            'http://www.someuri.com'
+        );
+
+        $this->assertEquals(
+            json_decode($this->successBody, true),
+            $uploader->resume('http://some-resume-uri.com')
+        );
+    }
+
+    public function testResumeFinishedUpload()
+    {
+        $statusResponse = new Response(200, [], $this->successBody);
+
+        $this->requestWrapper->send(
+            Argument::type('Psr\Http\Message\RequestInterface'),
+            Argument::type('array')
+        )->willReturn($statusResponse);
+
+        $uploader = new ResumableUploader(
+            $this->requestWrapper->reveal(),
+            $this->stream,
+            'http://www.someuri.com'
```

Comment:
Can you use `example.com` base domain?

e.g.
- someuri.example.com
- some-resume-uri.example.com